docs/upgrading/upgrading_to_v1.md

```diff
@@ -5,6 +5,95 @@ title: Upgrading to v1
 
 This page summarizes the breaking changes between Crawlee for Python v0.6 and v1.0.
 
+## ServiceLocator changes
```

Comment:
Could we move this section after "New architecture of storage clients" and before "Other smaller updates"? The service locator is more of an edge component and isn't used that often.